### PR TITLE
Add support for GitHub pages

### PR DESCRIPTION
**Pull request by pnmadelaine**
- add an action to push build results to GitHub
- add an action to deploy a job to GitHub pages

### CODE REVIEW

**Comment by pnmadelaine** [comment]
Converted to draft as this requires adjustments to work with custom domains

File: nix/lib/github/mkPushResult.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        cp -r "''${input[out]}" out
+        chmod -R +w out
+        cd out
+
+        git init
+        git config user.email "typhon@typhon-ci.org"
+        git config user.name "Typhon"
+        export GIT_AUTHOR_DATE="1970-01-01T00:00:00+0000"
+        export GIT_COMMITTER_DATE="1970-01-01T00:00:00+0000"
```
Comment (W95Psp) [inline nix/lib/github/mkPushResult.nix:30]: Would be nice to put that in utils, what do you think?

**Review by W95Psp — APPROVED** [review]

File: nix/lib/github/mkPushResult.nix (hunk 1 above)
Comment (pnmadelaine) [reply nix/lib/github/mkPushResult.nix:30]: actually it would be even nicer to have `lib.builders.mkActionScript` set this up, in the spirit of #38

Comment (pnmadelaine) [reply nix/lib/github/mkPushResult.nix]: this is beyond the scope of this PR anyway, so for the time being I moved this to `lib.git.config`

**Pull request merged**